lib/oneview-sdk/resource/api500/c7000/volume.rb

```diff
@@ -127,6 +127,29 @@ def self.add(client, storage_system, volume_name, is_shareable = false, options
           new(client, client.response_handler(response))
         end
 
+        # Retrieve resource details based on this resource's name or URI.
+        # @note one of the UNIQUE_IDENTIFIERS, e.g. name or uri or properties['name'], must be specified in the resource
+        # @return [Boolean] Whether or not retrieve was successful
+        def retrieve!
+          return super unless @data['properties']
+          results = find
+          if results.size == 1
+            set_all(results.first.data)
+            return true
+          end
+          false
+        end
+
+        # Check if a resource exists
+        # @note one of the UNIQUE_IDENTIFIERS, e.g. name or uri or properties['name'], must be specified in the resource
+        # @return [Boolean] Whether or not resource exists
+        def exists?
```

Comment:
This method does same thing that `retrieve!` method, right?
Maybe can be good call the `retrieve!` method here (Just do not use the self instance to call `retrieve!`, you can use a temp variable to that), avoiding write the "same" code.

Reply:
In that case I think not, because ```exists?``` does not set the resource. Perhaps the opposite, the ```exists?``` being called within ```retrieve!```.

Reply:
Using `exists?` within retrieve is likely the better alternative indeed

Reply:
I can see now. If I want to reuse only two first rows of each method. I need to call ```retrieve!``` inside ```exists?```, because I need a result to set up the data. Can I use the Ricardo's suggestion, which uses a temporary variable to avoid setting the data when ```exists?``` is called. I don't see big gains from this change. But it's okay, I'll do this and see what the code will look like.

Reply:
If you do not see big gains, keep the code as is it. No problem for me. =)
Important is the behavior to be ok.

Reply:
> I don't see big gains from this change. But it's okay, I'll do this and see what the code will look like.

Ops, true. Since `exists?` will only return a boolean instead of the resource, this is actually bad.